src/lib/yang/schema.lua

```diff
@@ -564,6 +570,26 @@ local primitive_types = set(
    'binary', 'bits', 'boolean', 'decimal64', 'empty', 'enumeration',
    'identityref', 'instance-identifier', 'leafref', 'string', 'union')
 
+-- Inherits config attributes from parents
+local function inherit_config(schema, env)
+   if env == nil then env = {env={}} end
+
+   if schema.config ~= nil then
+      env.config = schema.config
+   elseif env.env.config ~= nil then
+      schema = shallow_copy(schema)
+      schema.config = env.env.config
+   end
+
+   if schema.body then
```

Comment:
so let's say you have `local node = { config = { foo=bar } }`.  then `local node2 = shallow_copy(node1)`.  now node is not node2, but `node2.config == node.config`.  it's to fix this latter aliasing that the second `shallow_copy` is needed, I think